app/scripts/app.js

```diff
@@ -35,6 +35,14 @@ angular
     'openshiftCommonUI',
     'webCatalog'
   ])
+  .run(function(AuthService, HomePagePreference) {
+    HomePagePreference.gotoHomePage();
```

Comment:
yes it does :-) sorry.  So we only want to go to the home page when path is empty.  `$location.$$url` seems to contain this info; path minus the baseHref:

```
When https://localhost:9000/dev-console/project/myproject/browse/stateful-sets/
$location.$$url =  /project/myproject/browse/stateful-sets/

When https://localhost:9000/dev-console/
$location.$$url = "/"
```
Seem reasonable? Unless there is a helper function which already tests this.
Thanks,
Dave

Reply:
Added code to only go to home page when path is empty.

Reply:
Arrggg...now if you set your home page pref to 'project-list' and log out and back in, it doesn't redirect to '/projects' even though the $location.path('projects') is called.   

As I test I have the code below, but even though `$location.path('projects')` is called during `onUserChanged` browser doesn't redirect to the Project List.

```
.run(function(AuthService, HomePagePreference, $location) {
    if ($location.$$url === "/") {
      HomePagePreference.gotoHomePage();
    }
    AuthService.onUserChanged(function(user) {
      if (user) {
        //HomePagePreference.gotoHomePage();
        $location.path('projects');
      }
    });
  })
```

It's weird because when log in to project which no longer exists, you do get the toast notification, **and a redirect to Project List!**.   Log in/out works fine with Catalog(Home) and Project Overview set as home page pref, just not the project-list.

Reply:
Ok, had to add a `$timeout` [here](https://github.com/openshift/origin-web-console/pull/2562/files#diff-f9201982127fd8efc3606dcbb9fa087fR110) in order to navigate.